### PR TITLE
Fix _parse_date code so that it expects a path to a date string, rather than to a date element (v2.03)

### LINKED ISSUE #31: Incorrect date_order XPaths in IATI Standard Rulesets
The [IATI ruleset specification](http://iatistandard.org/104/rulesets/ruleset-spec/
) states that, for the `date_order ` rule:

> The date matched by less must be less than the date matched by more. If either of these dates is not found, the rule is ignored.
 
This `date_order ` rule is implemented in the IATI Standard ruleset [here](https://github.com/IATI/IATI-Rulesets/blob/version-2.02/rulesets/standard.json#L11-L14) (same rule at versions 1.04 to 2.02):

```
        "date_order": {
            "cases": [
                { "less": "activity-date[@type='1']", "more": "activity-date[@type='3']" },
                { "less": "activity-date[@type='2']", "more": "activity-date[@type='4']" },
                { "less": "activity-date[@type='2']", "more": "NOW" },
                { "less": "activity-date[@type='4']", "more": "NOW" }
            ]
        },
```
However, in all cases the stated XPaths for `less` and `more` are incorrect, since these are do not relate to that path where the dates are entered. 

To fix, each instance should be suffixed by the `@iso-date` attribute.

This issue appears to applies to all other instances where dates are referred to, i.e. these rules in the [v2.02 ruleset](https://github.com/IATI/IATI-Rulesets/blob/version-2.02/rulesets/standard.json#L45-L108).

### PR DESCRIPTION
**Pull request by andylolz**
Addresses https://github.com/IATI/IATI-Rulesets/issues/31#issuecomment-395060938:

> While standard.json was updated for all versions to match [the signature pyIATI expects](https://github.com/IATI/pyIATI/blob/59588100911f4fd17a4012c0c2bc9632cc20efbd/iati/rulesets.py#L573) (i.e. a path to a date _*string*_), the relevant code that actually runs the ruleset tests in this repo (that [expects a path to a date _*element*_](https://github.com/IATI/IATI-Rulesets/blob/e77e6860ceba3be71d95c7ed95b9c78f3f29928e/iatirulesets/__init__.py#L40)) wasn’t updated. So this change breaks that.
> 
> The metatests on this repo weren’t updated either, so the green tick on these various PRs is misleading.
> 
> ~~This bug causes lots of errors for IATI-Stats, so it’s worth addressing.~~ UPDATE: Since IATI/IATI-Stats#128, IATI-Stats no longer runs ruleset tests, so it’s no longer affected by this bug.

### CODE REVIEW

**Comment by andylolz** [comment]
I faffed about with the PHP a bit, but tests are now passing so I guess this is okay.

**Comment by Ocre42** [comment]
Hi @andylolz, given that pyIATI is no longer being worked on, do you think these PRs are still relevant or useful? If you still see these changes as something useful please let me know.

**Comment by andylolz** [comment]
@Ocre42 apologies – I probably should have done a better job of explaining what the bug is here / providing a reproducible error, rather than simply tacking a confusing comment onto #31.

I’ve now added a better explanation in #82.

tl;dr: This change was made as something to do with pyIATI (not sure what) but it doesn’t actually relate to pyIATI at all, and it’s probably still relevant if you want to get this repo working.
